### PR TITLE
feat: main screen popular tags components

### LINKED ISSUE #3: Main Screen - Popular Tags Component
Some popular tags to save time when people want to search for common topics

### PR DESCRIPTION
**Pull request by ulissesferreira**
Popular tags component of the main screen is now working. Front end routing is not yet fully implemented so part of it is not done yet since this is basically a redirect... Will start working on it ASAP

Ready for review and merge 💯 

Resolves #3

### CODE REVIEW

**Comment by ulissesferreira** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 12
           

Complexity increasing per file
==============================
- src/components/MainPopularTags.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005&bid=9683079)

File: src/components/MainPopularTags.js [hunk 1]
```diff
@@ -1,9 +1,41 @@
 import React, { Component } from 'react';
+import { Link } from 'react-router-dom'
```
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:2]: Codacy Issue found: [Strings must use doublequote. (quotes)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

File: src/components/MainPopularTags.js [hunk 2]
```diff
@@ -1,9 +1,41 @@
...
-    return (
-      <h1>Sou umas Popular Tags</h1>
-    );
+    if (this.state.tags !== []) {
+      return (
+        <div>
+          {this.state.tags.map(tag => {
+            return (<div><Link to={'/search/' + tag}>
+              {tag}
+            </Link></div>)
```
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:29]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

File: src/components/MainPopularTags.js [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
 
-export default class MainPopularTags extends Component {
   render() {
-    return (
-      <h1>Sou umas Popular Tags</h1>
-    );
+    if (this.state.tags !== []) {
+      return (
+        <div>
+          {this.state.tags.map(tag => {
```
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:26]: Codacy Issue found: [Expected parentheses around arrow function argument. (arrow-parens)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

File: src/components/MainPopularTags.js [hunk 4]
```diff
@@ -1,9 +1,41 @@
...
+            return (<div><Link to={'/search/' + tag}>
+              {tag}
+            </Link></div>)
+          })}
+        </div>
+      )
```
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:32]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

File: src/components/MainPopularTags.js [hunk 5]
```diff
@@ -1,9 +1,41 @@
 import React, { Component } from 'react';
+import { Link } from 'react-router-dom'
+import * as Api from '../utils/Api'
+
+export default class MainSearchBar extends Component {
+
+  constructor(props) {
+    super(props)
```
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:8]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

File: src/components/MainPopularTags.js [hunk 6]
```diff
@@ -1,9 +1,41 @@
...
+    this.state = {
+      tags : []
+    }
```
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:11]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

File: src/components/MainPopularTags.js [hunk 7]
```diff
@@ -1,9 +1,41 @@
 import React, { Component } from 'react';
+import { Link } from 'react-router-dom'
+import * as Api from '../utils/Api'
```
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:3]: Codacy Issue found: [Strings must use doublequote. (quotes)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

File: src/components/MainPopularTags.js [hunk 8]
```diff
@@ -1,9 +1,41 @@
...
+    this.state = {
+      tags : []
+    }
+  }
+
+  componentDidMount() {
+    Api.getSearchTags().then(res => {
+      this.setState({
+        tags : res
+      })
```
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:18]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

File: src/components/MainPopularTags.js (hunk 7 above)
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:3]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

File: src/components/MainPopularTags.js [hunk 9]
```diff
@@ -1,9 +1,41 @@
...
-export default class MainPopularTags extends Component {
   render() {
-    return (
-      <h1>Sou umas Popular Tags</h1>
-    );
+    if (this.state.tags !== []) {
+      return (
+        <div>
+          {this.state.tags.map(tag => {
+            return (<div><Link to={'/search/' + tag}>
```
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:27]: Codacy Issue found: [Strings must use doublequote. (quotes)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

File: src/components/MainPopularTags.js [hunk 10]
```diff
@@ -1,9 +1,41 @@
...
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      tags : []
+    }
+  }
+
+  componentDidMount() {
+    Api.getSearchTags().then(res => {
```
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:15]: Codacy Issue found: [Expected parentheses around arrow function argument. (arrow-parens)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

File: src/components/MainPopularTags.js (hunk 1 above)
Comment (ulissesferreira) [inline src/components/MainPopularTags.js:2]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/app/Upframe/connect/pullRequest?prid=2537005)

**Comment by fabiofcferreira** [comment]
I think we have to solve Codacy Quality Review problems before rebasing and merging this branch to master.

Thoughts?

**Comment by ulissesferreira** [comment]
I can solve those that make sense but some of them are dumb. I will refactor this so that most of the problems get fixed without compromising the create-react 
-app inbuilt linter.

**Review by fabiofcferreira — APPROVED** [review]

**Pull request merged**